### ISSUE TITLE
Bug: In VSCode, when an incorrect config breaks the ESLint server, the "Restart ESLint Server" command does not work

### ISSUE THREAD

**Issue by ppseprus** [eslint/eslint#18592]
### Environment

Node version: 20.12.2
npm version: 10.5.2
Local ESLint version: 9.5.0
**VSCode extension version: 2.4.4**
Global ESLint version: none
VSCode version: 1.90.1
Operating System: macOS Sonoma 14.5 (23F79)

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
// eslint.config.js
export default [{
  rules: {
    "no-unused-vars": "error",
  }
}]
```
</details>

# Steps to Reproduce
1. Using the above minimal config, have the ESLint server up and running in VSCode
```log
[Info  - 10:43:36 PM] ESLint server is starting.
[Info  - 10:43:37 PM] ESLint server running in node v20.9.0
[Info  - 10:43:37 PM] ESLint server is running.
[Info  - 10:43:37 PM] ESLint library loaded from: /Users/[...]/node_modules/eslint/lib/api.js
```
2. Break the config by misconfiguring it (eg. add `ignore` instead of `ignores`) **and hit save**
```log
[Error - 10:47:47 PM] An unexpected error occurred:
[Error - 10:47:47 PM] ConfigError: Config (unnamed): Unexpected key "ignore" found.
```
3. Remove / fix the incorrect **and hit save**
4. Open the Command Palette (cmd + shift + P) and select "ESLint: Restart ESLint Server"
5. Open the Command Palette (cmd + shift + P) and select "ESLint: Show Output Channel"

### What did you expect to happen?

The ESLint plugin to restart and work correctly.

### What actually happened?

After the restart, the ESLint server exits.

```log
...
[Error - 10:49:40 PM] Server process exited with code 0.
```

The developer has to reload VSCode to get it working again.

### Link to Minimal Reproducible Example

https://github.com/ppseprus/vscode-eslint-bug

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The same thing happens, when the config is provided via a `*.json` file and there's a trailing comma at the last line. Incorrectly formatted `*.json` breaks the server and the restart does not seem to work event after the issue is fixed.

**Comment by ppseprus** [comment]
I just realised that the ESLint server says its **"running in node v20.9.0"**, while I use `nvm` and I don't even have that version installed. As I noted above, I have v20.12.2 installed... 🤔

**Comment by nzakas** [comment]
We don't maintain the VS Code extension. You'll need to open an issue there:
https://github.com/Microsoft/vscode-eslint

**Comment by ppseprus** [comment]
Ah! Sorry! My bad! I have no idea how I mixed up the two repos! 😞

**Comment by ppseprus** [comment]
In case someone lands on this page —> https://github.com/microsoft/vscode-eslint/issues/1873
